### DISCUSSION TITLE
Filter options and results not matching

### DISCUSSION THREAD

**Discussion by BVSanjay** [microsoft-search/pnp-modern-search#3694] [General]
**Version used**
Ex: 4.11.0.0

I am doing documents search with query to fetch only documents with filter isDocument:1 and Result source is Local SharePoint Results

When I search for some particular term I get 4 files of extension 'eml'. But in filters I get two options 'csv' and 'eml' even though the results are only eml files.
When we select filter as 'csv', I get 3 new CSV files which did not show up before. Again if I clear filter I get the old 4 'eml' files 

**Expected behavior**
Show all files

**Comment by kasperbolarsen** [comment]
Please retest the scenario using the most resent version.  We can't support anything but the current version

**Comment by BVSanjay** [comment]
It is same in version 4.11.0

**Comment by kasperbolarsen** [comment]
Try using SP Editor or the SharePoint Search query tool to inspect the results returned by the API. 
It sounds very much like an issue with the display values vs the the values in the Refiner, which crawled property is it you are using in the RefinableString that is used in the Filter ?

**Comment by wobba** [comment]
IsDocument:1 is not always useful in my opinion. Can you explain what specific files you want returned, or all files? If all files, then use something like this instead `(contentclass:STS_ListItem_DocumentLibrary OR contentclass:STS_ListItem_MySiteDocumentLibrary OR contentclass:STS_ListItem_PictureLibrary OR contentclass:STS_ListItem_ModernTemplateLibrary OR contentclass:STS_ListItem_851)`
